File: src/repl.ts
import * as readline from 'readline';
import { Digits, PersianHex } from './persian_hex/persian_hex';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function main() {
    rl.question('', (user_input: string) => {
        try {
            const number = parseInt(user_input.trim());
            if (isNaN(number) || number < 0) {
                console.log('Error: Please enter a valid non-negative integer.');
                return;
            }

            const persianHex = new PersianHex();
            persianHex.setMode(Digits.ENGLISH);
            const result = persianHex.calculate(number);
            console.log(result);
        } catch (error) {
            console.log(`Error: ${error.message}`);
        } finally {
            rl.close();
        }
    });
}

main();

File: src/repl.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String userInput = scanner.nextLine();

        try {
            int number = Integer.parseInt(userInput.trim());
            if (number < 0) {
                System.out.println("Error: Please enter a valid non-negative integer.");
                return;
            }

            PersianHex persianHex = new PersianHex();
            persianHex.setMode(Digits.ENGLISH);
            String result = persianHex.calculate(number);
            System.out.println(result);
        } catch (NumberFormatException e) {
            System.out.println("Error: Invalid number format.");
        }
    }
}
